src/lib/handle_ping_start.go

```diff
@@ -0,0 +1,86 @@
+package lib
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/zairza-cetb/bench-routes/src/lib/filters"
+	"github.com/zairza-cetb/bench-routes/src/lib/modules/ping"
+	"github.com/zairza-cetb/bench-routes/src/lib/utils"
+	"github.com/zairza-cetb/bench-routes/tsdb"
+)
+
+func handlePingStart(config utils.YAMLBenchRoutesType, pingServiceState string) bool {
+	if pingServiceState == "passive" {
+
+		Configuration.Config.UtilsConf.ServicesSignal.Ping = "active"
+		_, e := Configuration.Write()
+		if e != nil {
+			panic(e)
+		}
+		go func() {
+			pingConfig := Configuration.Config.Routes
+			pingInterval := getInterval(Configuration.Config.Interval, "ping")
+			if pingInterval == (testInterval{}) {
+				log.Fatalf("interval not found in configuration file for ping")
+				return
+			}
+
+			urlStack := make(map[string]string)
+			urlHashMap := make(map[string]string)
```

Comment:
thanks

Reply:
@Harkishen-Singh I would most likely require this logic for having unique routes universal. For example, we need unique routes for every module which have different inputs in each case(say `route url` for ping module, `route url and method` for req-res delay). So, the filtering must be done on the route itself(not the url) which means store the routes in some Data Structure and filter based on route.URL. Thoughts?

Reply:
Basically, the idea is to make a routeStack not a urlStack and append/delete based on route.URL. Why decompose the route struct as per URL which seems to be a `variant` property in comparision to the route struct, which is `invariant`

Reply:
I understand your point @aquibbaig. `urlStack` is used here since ping is concerned with unique IP addresses than unique routes. however, for most modules, we can also maintain a global `routeStack` as you mentioned.

However, pls note that the `routeStack` will contain IP addresses too, due to the case of different VM instances having different routes required to be tested as well.

Reply:
This can be done in a different PR as this PR is specific to **simplification**. if possible, pls try having the above idea in req-res PR.

Thanks!